vtex-sites/storecomponents.store
# Use thumbor for carousel

## What's the purpose of this pull request?
As title says, this PR adds gatsby-plugin-image and gatsby-plugin-thumbor to this store. Also, as a migration strategy, we are migrating the carousel to use this new image plugin.

## How it works? 
Image quality and performance should not be a compromise the headless CMS operator should choose between. The framework should offer a good, out of the box, image processing service that offers a good tradeoff between performance and image quality. If, however, the images are a core business of the store and headless CMS operators wants to have this responsibility, the framework should allow these operators a fine grained control over image settings.

This PR integrates with a new image processing service called [thumbor](http://thumbor.org/), an open-source smart on-demand image cropping, resizing and filters, via the great [gatsby-plugin-image](https://www.gatsbyjs.com/plugins/gatsby-plugin-image/). 
The integration is done via a url-based CDN optimization and is handled by `@vtex/gatsby-plugin-thumbor`.

This PR imports and uses components exported by both `gatsby-plugin-image` and `@vtex/gatsby-plugin-thumbor` to refactor the Carousel component. This refactoring introduces two new concepts files, `styles.ts`  and `image.config.js`.

Looking closer to the carousel component implementation you will now see that we have split the carousel's component into two files, `index.tsx`, that has the component's structure and logic, and `styles.json`, that have all of the component's css.
We are doing this because the carousel component is only used in a single page on the store, and having it to have variants is not really necessary anymore. Variants only introduce unnecessary bloat to the stores because of their global scope. Since we had to move the carousel's code here because of the image component change, we are already making this style change

The second concept is the `image.config.js` file. This file contains image variants and their respective configuration. This file is important because `@vtex/gatsby-plugin-thumbor` needs to know the sizes of all images being used on the store. We could only give the plugin a list of used sizes, but then every time we change the size of an image, we would need to change the plugin config. With `image.config.js`, we can automate this process, with the drawback being that you need to change images config in a centralized file instead of where the image is used

## How to test it?
Open the deploy preview and make sure the banner image is:
1. Showing up on different screens/browsers
2. Notice a size reduction in comparison with the live version

## Results
Image sizes before and after the introduction of the thumbor server
Laptop with HiDPI : 395Kb => 248Kb
Mobile Moto G4: 106Kb => 65Kb
